### PR TITLE
Fix outdated parameter cache

### PR DESCRIPTION
**Pull request by Pessimistress**
For #1509

#### Change List
- Do not cache unreferenced parameters
- Unit test

### CODE REVIEW

File: modules/gltools/src/state-tracker/track-context-state.ts [hunk 1]
```diff
@@ -19,10 +19,10 @@ function installGetterOverride(gl, functionName) {
   gl[functionName] = function get(...params) {
     const pname = params[0];
 
-    // WebGL limits are not prepopulated in the cache, we must
-    // query first time. They are all primitive (single value)
+    // WebGL limits are not prepopulated in the cache, it's neither undefined in GL_PARAMETER_DEFAULTS
+    // nor intercepted by GL_HOOKED_SETTERS. Query the original getter.
     if (!(pname in gl.state.cache)) {
-      gl.state.cache[pname] = originalGetterFunc(...params);
+      return originalGetterFunc(...params);
```
Comment (ibgreen) [inline modules/gltools/src/state-tracker/track-context-state.ts:25]: Hmm, at first look it seems this fixes the issue by disabling all caching?

**Review by ibgreen — APPROVED** [review]

File: modules/gltools/src/state-tracker/track-context-state.ts (hunk 1 above)
Comment (Pessimistress) [reply modules/gltools/src/state-tracker/track-context-state.ts:25]: `cache` is initialized with every parameter in GL_PARAMETER_DEFAULTS: https://github.com/visgl/luma.gl/blob/0b530bc91b6c76df8ca5417bdc63032a343e5419/modules/gltools/src/state-tracker/track-context-state.ts#L106

Additional keys may be added by calling an intercepted setter.

After this change, if a parameter should always be cached, it needs to be added to the GL_PARAMETER_DEFAULTS table. Which is what we are already doing for the most commonly used gl states.

Comment (ibgreen) [reply modules/gltools/src/state-tracker/track-context-state.ts:25]: I suppose we get a perf hit for all the static limits queries but that is probably OK as they are usually not queried very often. We could add a list of static values to address that.

Comment (Pessimistress) [reply modules/gltools/src/state-tracker/track-context-state.ts:25]: Created issue https://github.com/visgl/luma.gl/issues/1511

**Pull request merged**